In [1]:
import pandas as pd

# Usando o método `read_csv` para ler um arquivo .csv
Primeiramente vamos ler o um arquivo csv chamado `pokemon.csv` usando a função:

In [4]:
pd.read_csv("../data/pokemon.csv")

Pokemon     Type
0     Bulbasaur    Grass
1       Ivysaur    Grass
2      Venusaur    Grass
3    Charmander     Fire
4    Charmeleon     Fire
..          ...      ...
716     Yveltal     Dark
717     Zygarde   Dragon
718     Diancie     Rock
719       Hoopa  Psychic
720   Volcanion     Fire

[721 rows x 2 columns]

- A função `read_csv` tem dezenas de parâmetros. Para entender cada um deles, temos que recorrer a [documentação](https://pandas.pydata.org/docs/reference/api/pandas.read_csv.html)
- Perceba que a resposta do método foi uma tabela com duas colunas, que difere de uma série
- Na verdade, essa tabela se chama `DataFrame`, mas vamos falar sobre ele em breve
- Por agora, vamos carregar apenas uma coluna para termos uma série. Para isso, vamos usar:
    - O parâmetro `usecols` e passar apenas a coluna Pokemon (assim, temos uma unica coluna)
    - O parâmetro `squeeze` para forçar nossa dataframe se tornar uma série

In [6]:
pokemon = pd.read_csv("../data/pokemon.csv", usecols=["Pokemon"]).squeeze()
pokemon

0       Bulbasaur
1         Ivysaur
2        Venusaur
3      Charmander
4      Charmeleon
          ...    
716       Yveltal
717       Zygarde
718       Diancie
719         Hoopa
720     Volcanion
Name: Pokemon, Length: 721, dtype: object

- Observe que temos 721 pokemons (linhas)
- Porém, o Pandas está mostrando apenas 10. Isso é apenas para economizar espaço e não explodir nossa tela
- Podemos dar uma olhada apenas no inicio e no fim da serie usando os métodos `head` and `tail`

In [7]:
pokemon.head(10) # Monstra as 10 primeiras linhas. Se nao passarmos o 10, o padrão é 5

0     Bulbasaur
1       Ivysaur
2      Venusaur
3    Charmander
4    Charmeleon
5     Charizard
6      Squirtle
7     Wartortle
8     Blastoise
9      Caterpie
Name: Pokemon, dtype: object

In [8]:
pokemon.tail(3) # Monstra as 3 últimas linhas. Se nao passarmos o 3, o padrão é 5

718      Diancie
719        Hoopa
720    Volcanion
Name: Pokemon, dtype: object

In [9]:
pokemon.tail()

716      Yveltal
717      Zygarde
718      Diancie
719        Hoopa
720    Volcanion
Name: Pokemon, dtype: object

## Usando funções padrões do Python com a serie
- Podemos usar `len`, `type`, `sorted`, `list`, `max`, `min`, etc

In [10]:
len(pokemon)

721

In [11]:
type(pokemon)

pandas.core.series.Series

In [12]:
sorted(pokemon)

['Abomasnow',
 'Abra',
 'Absol',
 'Accelgor',
 'Aegislash',
 'Aerodactyl',
 'Aggron',
 'Aipom',
 'Alakazam',
 'Alomomola',
 'Altaria',
 'Amaura',
 'Ambipom',
 'Amoonguss',
 'Ampharos',
 'Anorith',
 'Arbok',
 'Arcanine',
 'Arceus',
 'Archen',
 'Archeops',
 'Ariados',
 'Armaldo',
 'Aromatisse',
 'Aron',
 'Articuno',
 'Audino',
 'Aurorus',
 'Avalugg',
 'Axew',
 'Azelf',
 'Azumarill',
 'Azurill',
 'Bagon',
 'Baltoy',
 'Banette',
 'Barbaracle',
 'Barboach',
 'Basculin',
 'Bastiodon',
 'Bayleef',
 'Beartic',
 'Beautifly',
 'Beedrill',
 'Beheeyem',
 'Beldum',
 'Bellossom',
 'Bellsprout',
 'Bergmite',
 'Bibarel',
 'Bidoof',
 'Binacle',
 'Bisharp',
 'Blastoise',
 'Blaziken',
 'Blissey',
 'Blitzle',
 'Boldore',
 'Bonsly',
 'Bouffalant',
 'Braixen',
 'Braviary',
 'Breloom',
 'Bronzong',
 'Bronzor',
 'Budew',
 'Buizel',
 'Bulbasaur',
 'Buneary',
 'Bunnelby',
 'Burmy',
 'Butterfree',
 'Cacnea',
 'Cacturne',
 'Camerupt',
 'Carbink',
 'Carnivine',
 'Carracosta',
 'Carvanha',
 'Cascoon',
 'Castform',


In [13]:
list(pokemon)

['Bulbasaur',
 'Ivysaur',
 'Venusaur',
 'Charmander',
 'Charmeleon',
 'Charizard',
 'Squirtle',
 'Wartortle',
 'Blastoise',
 'Caterpie',
 'Metapod',
 'Butterfree',
 'Weedle',
 'Kakuna',
 'Beedrill',
 'Pidgey',
 'Pidgeotto',
 'Pidgeot',
 'Rattata',
 'Raticate',
 'Spearow',
 'Fearow',
 'Ekans',
 'Arbok',
 'Pikachu',
 'Raichu',
 'Sandshrew',
 'Sandslash',
 'Nidoran',
 'Nidorina',
 'Nidoqueen',
 'Nidoran♂',
 'Nidorino',
 'Nidoking',
 'Clefairy',
 'Clefable',
 'Vulpix',
 'Ninetales',
 'Jigglypuff',
 'Wigglytuff',
 'Zubat',
 'Golbat',
 'Oddish',
 'Gloom',
 'Vileplume',
 'Paras',
 'Parasect',
 'Venonat',
 'Venomoth',
 'Diglett',
 'Dugtrio',
 'Meowth',
 'Persian',
 'Psyduck',
 'Golduck',
 'Mankey',
 'Primeape',
 'Growlithe',
 'Arcanine',
 'Poliwag',
 'Poliwhirl',
 'Poliwrath',
 'Abra',
 'Kadabra',
 'Alakazam',
 'Machop',
 'Machoke',
 'Machamp',
 'Bellsprout',
 'Weepinbell',
 'Victreebel',
 'Tentacool',
 'Tentacruel',
 'Geodude',
 'Graveler',
 'Golem',
 'Ponyta',
 'Rapidash',
 'Slowpoke',
 'Slo

In [14]:
max(pokemon)

'Zygarde'

In [15]:
min(pokemon)

'Abomasnow'